stablehlo/reference/InterpreterOps.cpp

```diff
@@ -193,17 +193,17 @@ SmallVector<InterpreterValue> evalRunParallelOp(
   return results;
 }
 
+// `serializedProbeFileId` should be a unique positive integer which can be used
+// to unambiguously derive a serialized filename for a given `probeId`.
 llvm::Error evalProbeOp(InterpreterValue input, StringRef probeId,
                         StringRef probeOutputDir,
-                        llvm::StringMap<int32_t> &probeIterations) {
+                        int64_t serializedProbeFileId) {
   llvm::SmallString<128> filepath(probeOutputDir);
 
-  // To properly support loops, append a suffix denoting how many times this
-  // specific probe_id has executed.
-  const int32_t numTimesExecuted = ++probeIterations[probeId];
-
+  // Use an increasing unique integer to write to disk to avoid any odd file
+  // names as a result of unsafe probe_id values.
   llvm::sys::path::append(
-      filepath, probeId + "_" + std::to_string(numTimesExecuted) + ".npy");
+      filepath, "probe" + std::to_string(serializedProbeFileId) + ".npy");
```

Comment:
Is this preferable to enforcing a grammar on probe IDs?

Can add [let hasVerifier = 1](https://github.com/openxla/stablehlo/blob/e5ad51715a11721c78b6748ab5de7945df24b1b8/stablehlo/reference/InterpreterOps.td#L80C1-L80C1) and implement a [verify](https://github.com/openxla/stablehlo/blob/e5ad51715a11721c78b6748ab5de7945df24b1b8/stablehlo/reference/InterpreterOps.cpp#L87) method which enforces `[A-Za-z_]+` for probe name.

There's something kind of nice about the probe filename matching the probe_id

Reply:
This is somewhat tangentially related to the conversation in https://github.com/openxla/stablehlo/pull/1841#discussion_r1393341371.

In short, I'm hesitant to enforce a grammar here as different compilers may populate MLIR location data differently, which could be used by the StableHLO probe pass to populate `probe_id` values in the proposed PR in #1841.